app/controllers/posts_controller.rb

```diff
@@ -2,6 +2,7 @@ class PostsController < ApplicationController
   def index
     @posts = Post.all
     @user = User.find(params[:id])
+    @user.includes(:posts, :comments)
   end
 
   def show
```

Comment:
- [ ] I had this error when I try to create a post and click on the see all posts, it is happened because of includes keyword, please fix it.

![validate2](https://user-images.githubusercontent.com/55635977/147197662-9a20dacc-fadb-433a-9b1f-8fb76ec89593.png)
![validate3](https://user-images.githubusercontent.com/55635977/147197676-620ec703-3ff9-4486-a0c3-52f91ee85021.png)


- [ ]  I see you are trying to solve the N+1 problem however, you would have implemented this on the `show method ` because The N+1 query problem happens when your code executes N additional query statements to fetch the same data that could have been retrieved when executing the primary query.

Reply:
Thank you @Emmyn5600 you are totally right, now I fixed the bug 🐛 and change the N+1 implementation in the show method 😄